sourcemap/fleet_store_test.go

```diff
@@ -77,4 +77,106 @@ func TestFleetFetch(t *testing.T) {
 	assert.True(t, hasAuth)
 }
 
+func TestMultipleFleetHostsFetch(t *testing.T) {
+	var (
+		hasAuth       bool
+		apikey        = "supersecret"
+		name          = "webapp"
+		version       = "1.0.0"
+		path          = "/my/path/to/bundle.js.map"
+		c             = http.DefaultClient
+		sourceMapPath = "/api/fleet/artifact"
+	)
+
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, sourceMapPath, r.URL.Path)
+		auth := r.Header.Get("Authorization")
+		hasAuth = auth == "ApiKey "+apikey
+		// zlib compress
+		wr := zlib.NewWriter(w)
+		defer wr.Close()
+		wr.Write([]byte(resp))
+	})
+
+	ts0 := httptest.NewServer(h)
+	defer ts0.Close()
+
+	ts1 := httptest.NewServer(h)
+	defer ts1.Close()
+
+	fleetCfg := &config.Fleet{
+		Hosts:        []string{ts0.URL[7:], ts1.URL[7:]},
+		Protocol:     "http",
+		AccessAPIKey: apikey,
+		TLS:          nil,
+	}
+
+	cfgs := []config.SourceMapMetadata{
+		{
+			ServiceName:    name,
+			ServiceVersion: version,
+			BundleFilepath: path,
+			SourceMapURL:   sourceMapPath,
+		},
+	}
+	f, err := newFleetStore(c, fleetCfg, cfgs)
+	assert.NoError(t, err)
+
+	gotRes, err := f.fetch(context.Background(), name, version, path)
+	require.NoError(t, err)
+
+	assert.Contains(t, gotRes, "webpack:///bundle.js")
+	assert.True(t, hasAuth)
+}
+
+func TestMultipleFleetHostsQueryFailureFetch(t *testing.T) {
+	var (
+		apikey        = "supersecret"
+		name          = "webapp"
+		version       = "1.0.0"
+		path          = "/my/path/to/bundle.js.map"
+		c             = http.DefaultClient
+		sourceMapPath = "/api/fleet/artifact"
+		requestCount  = 0
+	)
+
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "err", http.StatusInternalServerError)
+		requestCount++
```

Comment:
this should either use https://pkg.go.dev/sync/atomic#AddInt32, or a mutex; the requests are issued and handled concurrently